src/workerd/api/global-scope.c++

```diff
@@ -520,6 +552,21 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketError(
   KJ_IF_SOME(h, exportedHandler) {
     KJ_IF_SOME(handler, h.webSocketError) {
       auto promise = handler(js, kj::mv(websocket), js.exceptionToJs(kj::mv(e)));
+
+      // If we have a maximum event duration timeout set, we should prevent the actor from running
+      // for more than the user selected duration.
+      KJ_IF_SOME(timeout, eventTimeout) {
+        if (timeout > 0) {
+          // We must have an actor if we are delivering an hibernatable web socket message.
+          auto& actor = KJ_ASSERT_NONNULL(IoContext::current().getActor());
+
+          promise = promise.exclusiveJoin(IoContext::current().afterLimitTimeout(timeout * kj::SECONDS)
+            .then([&actor]() {
```

Comment:
Related to https://github.com/cloudflare/workerd/pull/628/files#r1445229771 above, can be turned to a coroutine?

Reply:
@jqmmes (tagging for visibility these things can be hard to find :upside_down_face:) this comment got resolved but it doesn't look like it was addressed, are we not able to make it a coroutine?

Reply:
Latest update addresses this :slightly_smiling_face: